engine/src/main/java/io/zeebe/engine/processing/job/FailProcessor.java

```diff
@@ -33,9 +38,21 @@ private void acceptCommand(
       final TypedRecord<JobRecord> command, final CommandControl<JobRecord> commandControl) {
     final long key = command.getKey();
     final JobRecord failedJob = state.getJob(key);
-    failedJob.setRetries(command.getValue().getRetries());
+    final long retryBackOff = command.getValue().getRetryBackOff();
+    final int retries = command.getValue().getRetries();
+    failedJob.setRetries(retries);
+    failedJob.setRetryBackOff(retryBackOff);
     failedJob.setErrorMessage(command.getValue().getErrorMessageBuffer());
-    state.fail(key, failedJob);
+    if (retryBackOff != 0) {
```

Comment:
If the remaining retries are less than one then we don't need to reschedule the job.

```suggestion
    if (retries > 0 && retryBackOff > 0) {
```

Reply:
Done